packages/compartment-mapper/src/parse-cjs-shared-export-wrapper.js

```diff
@@ -9,6 +9,13 @@ const { freeze, keys, create, hasOwnProperty } = Object;
  */
 const has = (object, key) => apply(hasOwnProperty, object, [key]);
 
+export const dropFileProtocol = url => {
+  if (url.substring(0, 7) === 'file://') {
+    url = url.substring(7);
+  }
+  return url;
+};
```

Comment:
This isn’t quite sufficient, accounting for Windows. The `readPowers` threaded into all of the Compartment Mapper API’s capture `url` (`url.filePathToURL`, `url.pathToFileURL`) so it can convert between URL space and path space.

Reply:
When importLocation is used, read powers don't enter compartment-mapper and even if I threaded them down to the parser implementation I couldn't leverage them. I can use the `new URL … url.pathname` implementation locally but don't see means of drilling the original reference to it down to the parser reliably.

Reply:
I think a reasonable fallback position is to provide `__dirname === undefined` if the compartment mapper doesn’t receive the `canonicalize` property on `ReadPowers`. You may have to change the signature to `ReadFn | ReadPowers` to thread it, but there are examples where I’ve already done this. There’s a utility function for normalizing that type laying around somewhere.

Reply:
readpowers is `{ read, canonical }`.   
`canonical` is not helpful for providing a file path on windows. I will expose existing path2url and url2path functions.